### DISCUSSION TITLE
Figure full refreshes on changing anything?

### DISCUSSION THREAD

**Discussion by contagon** [alandefreitas/matplotplusplus#143] [Q&A]
I'm currently trying to live plot an algorithm I'm developing, however, when updating the axis, the plot does a quick flash. This makes things unusable when updating too frequently. It also doesn't appear to matter what I'm changing - something as small as line thickness also seems to cause this.

```
int main() {
    using namespace matplot;
    int n = 10000;
    std::vector<double> x = linspace(0, 2 * pi, n);
    std::vector<double> y = transform(x, [](auto x) { return sin(x); });

    auto f = figure(false);
    auto ax = gca();
    auto p = ax->plot(x,y);
    f->draw();

    for(int i=10;i<n;i++){
        p->color("red").line_width(2);
        sleep(.1);
    }
    
    return 0;
}
```

Is there something I'm doing incorrectly here? Is this just a general limitation of matplotplusplus? I've tried a similar experiment using gnuplot directly, and I don't seem to get any of the refreshes.

Thanks! And thanks for all your work on this, I believe it's going to be my go-to plotting tool for C++.

**Comment by alandefreitas** [comment]
You're not doing anything incorrectly. It's a limitation with the backend, which is based on process pipes. You might try using another gnuplot terminal, but updating plots very frequently is unlikely to work without these flashes because the pipe is redrawing the whole figure. We're working on another backend but this is going to take a while.

**Reply by Haron123** [reply]
Is there already a way to live plot without the full figure refreshing currently ?

**Reply by alandefreitas** [reply]
Not really. Not a lot of work going into this library lately.

**Reply by OlClown** [reply]
I am having an adjacent issue I believe.  I have a figure and I update the plot based on user selections but on refresh, as discussed above, the whole plot flashes.

The flash is acceptable for my application but one issue is that the figure/window goes back to its original size when refreshed.  

I was thinking that I could grab the new size from the figure/backend(after the user changes the window size via dragging) and then after refreshing, set the size to the users desired size but when I call figure->height() or figure->width() it always just returns the starting size and does not seem to update with the changes being made to the figure window.  Am I hitting a limitation here or should what I am doing work?
